src/witan/send/adroddiad/vega_specs/lines.clj

```diff
@@ -327,18 +344,86 @@
+  "Converts a plot spec with `group`s identified by abbreviations to use labels:
+   Uses `:label` column from `colors-and-shapes` (if present) update `data` `group` `colors-and-shapes`
+   in a `plot-spec` to group by the labelled values.
+   If the `colors-and-shapes` dataset in the `plot-spec` contains a `:label` column:
+   - The (domain) values in the `group` column are mapped to the corresponding labels
+     according to the `colors-and-shapes` `:domain-value` -> `:label` mapping.
+   - The `colors-and-shapes` dataset is updated with the `:label`s as new `:domain-values`.
+   - If a `group-label` column is specified then the labels are put in that column,
+     the `group` key updated to refer to the new `group-label` column,
+     and the `group-label` key removed from the `plot-spec`."
+  [{:keys [data group colors-and-shapes group-label]
+    :or   {group-label nil}
+    :as   plot-spec}]
+  (let [group-label (or group-label group)]
+    (-> plot-spec
+        (dissoc :group-label)
+        (merge (when (some #{:label} (tc/column-names colors-and-shapes))
+                 (let [domain-value->label (as-> colors-and-shapes $ (zipmap ($ :domain-value) ($ :label)))]
+                   {:data              (as-> data $
+                                         (tc/map-columns $ group-label [group] #(get domain-value->label % %))
+                                         (tc/reorder-columns $ (concat (take-while (complement #{group}) (tc/column-names $))
+                                                                       [group group-label])))
+                    :group             group-label
+                    :colors-and-shapes (-> colors-and-shapes
+                                           (tc/drop-columns [:domain-value])
+                                           (tc/rename-columns {:label :domain-value}))}))))))
+
+(defn remove-tooltips
+  "Remove tooltips from a chart:
+   - Remove hover rules and associated `:transform` sub `:layer`s from a vl `chart`.
+     (These are added as tooltip by `line-and-ribbon-and-rule-plot`s.)
+   - Set any remaining `:tooltip` values to `nil`."
+  [chart]
+  (as-> chart $
+    ;; Remove "hover" rule layer
+    (update $ :layer (partial remove (fn [m] (and (some->> m :mark :type (= "rule"))
+                                                  (some->> m :params (some #(some->> % :name (= "hover"))))))))
+    ;; Remove any `:transform` layers within layers
+    (update $ :layer (partial mapv (fn [m] (if (contains? m :layer)
+                                             (update m :layer (partial remove #(contains? % :transform)))
+                                             m))))
+    ;; Disable any remaining `:tooltip`s by setting value to `nil`
+    (walk/prewalk (fn [e] (if (and (map-entry? e) (= (key e) :tooltip))
+                            [(key e) nil]
+                            e))
+                  $)))
+
+
+
+;;; # Wrappers
+(defn plot
```

Comment:
`plot-ehcps-by-year`?

I know the chart spec can over ride it, but the EHCP and line/ribbon assumptions are pretty baked in